### DISCUSSION TITLE
File copied to the clipboard when closing

### DISCUSSION THREAD

**Discussion by xxxxme** [gsantner/markor#2391] [Q&A]
On A13, Markor 2.12.5.154, when I click the save button, Markor tells me "Could not opn file." When I close the app, the file is saved anyway and the entire contents are copied to the clipboard. The app hasn't asked for any permissions. What do I need to do for this to work properly?

**Comment by harshad1** [comment]
This is a defensive mechanism that I implemented. It triggers if markor thinks the file is not written correctly.

Does this happen every time? Where are you opening the files from?

Not that you shouldn't need to press save. Markor will automatically save the file when it goes into the background or closes

**Comment by xxxxme** [comment]
Files are in a subfolder of Documents.
After more research, it seems this directory and its files are owned by "External Storage".  Files owned by "Media Storage" behave as expected.

**Reply by xxxxme** [reply]
Unfortunately, I don't seem to be able to change this, or else file owner immediately reverts when changed.

**Comment by harshad1** [comment]
Does markor have full filesystem permissions?

Android settings -> all files access

Markor should have asked for this on startup

**Comment by xxxxme** [comment]
It does have this permission.

**Reply by harshad1** [reply]
Hmm. How was the file created?

**Reply by xxxxme** [reply]
These files were extracted from a backup .zip with the file manager app I have installed on my phone. The zip was copied from another device using a USB cable.
New files created with both the file manager and Markor have "Media Storage" as owner.

**Comment by harshad1** [comment]
Hmmm. Is it possible for the file to not be writeable from markor? Are you rooted?

**Comment by xxxxme** [comment]
The phone is rooted.
It must be writeable *somehow* because the changes *are* saved when I close the app.
